lib/vector-core/src/event/metric/value.rs

```diff
@@ -2,51 +2,84 @@ use core::fmt;
 use std::collections::BTreeSet;
 
 use float_eq::FloatEq;
-use serde::{Deserialize, Serialize};
 use vector_common::byte_size_of::ByteSizeOf;
+use vector_config::configurable_component;
 
 use super::{samples_to_buckets, write_list, write_word};
 use crate::metrics::AgentDDSketch;
 
-#[derive(Debug, Clone, Deserialize, Serialize)]
+/// Metric value.
+#[configurable_component]
+#[derive(Clone, Debug)]
 #[serde(rename_all = "snake_case")]
 /// Container for the actual value of a metric.
 pub enum MetricValue {
     /// A simple value that can not decrease except to reset it to zero.
-    Counter { value: f64 },
+    Counter {
+        /// The value of the counter.
+        value: f64,
+    },
+
     /// A sampled numerical value.
```

Comment:
Is this actually a sampled value?

Reply:
No, this should be "A single numerical value that can arbitraril go up or down."

Will fix.

Reply:
Fixed in 8b55d4493. Also tried to make the wording of some of the other metric value kinds a bit more consistent.